state/upgrades.go

```diff
@@ -1884,3 +1885,54 @@ func UpdateInheritedControllerConfig(pool *StatePool) error {
 	}
 	return nil
 }
+
+// EnsureDefaultModificationStatus ensures that there is a modification status
+// document for every machine in the statuses.
+func EnsureDefaultModificationStatus(pool *StatePool) error {
+	st := pool.SystemState()
+	db := st.db()
+
+	machineCol, machineCloser := db.GetRawCollection(machinesC)
+	defer machineCloser()
+	// Find all machines that are alive.
+	machineIter := machineCol.Find(bson.D{{"life", life.Alive}}).Iter()
+	defer machineIter.Close()
+
+	statusCol, statusCloser := db.GetRawCollection(statusesC)
+	defer statusCloser()
+
+	var ops []txn.Op
+	var machine machineDoc
+	for machineIter.Next(&machine) {
+		key := machineGlobalModificationKey(machine.Id)
+
+		rawDoc := statusDoc{
+			ModelUUID:  machine.ModelUUID,
+			Status:     status.Idle,
+			StatusInfo: "",
+			StatusData: make(map[string]interface{}),
+			Updated:    st.clock().Now().UnixNano(),
+		}
+
+		// We only need to migrate machines that don't have a modification
+		// status document. So we need to first check if there is one, before
+		// creating a txn.Op for the missing document.
+		var status statusDoc
+		err := statusCol.Find(bson.D{{"_id", key}}).Select(bson.D{{"_id", 1}}).One(&status)
+		if err == mgo.ErrNotFound {
```

Comment:
his would probably be easier to read if you move this query earlier and just have

if err == nil {
  continue
} else if err != mgo.ErrNotFound {
  return errors.Trace(err)
}

then you don't have to nest the logic and you don't have to create rawDoc if you aren't going to use it.